Client/Telemetry/TelemetryWrapper.swift

```diff
@@ -1854,6 +1859,33 @@ extension TelemetryWrapper {
                 let properties = GleanMetrics.AppErrors.LargeFileWriteExtra(size: quantity)
                 GleanMetrics.AppErrors.largeFileWrite.record(properties)
             }
+        case(.action, .tap, .fxSuggest, _, let extras ):
+            guard let contextIdString = TelemetryContextualIdentifier.contextId,
+                  let contextId = UUID(uuidString: contextIdString),
+                  let interactionInfo = extras?[EventValue.fxSuggestionClickInfo.rawValue] as? RustFirefoxSuggestionInteractionInfo else { return }
+            switch interactionInfo {
+                case .amp(blockId: let blockId, advertiser: let advertiser, iabCategory: let iabCategory, reportingURL: let reportingURL):
+                    GleanMetrics.FxSuggest.contextId.set(contextId)
+                    GleanMetrics.FxSuggest.pingType.set("fxsuggest-click")
+                    GleanMetrics.FxSuggest.blockId.set(blockId)
+                    GleanMetrics.FxSuggest.advertiser.set(advertiser)
+                    GleanMetrics.FxSuggest.iabCategory.set(iabCategory)
+                    if let reportingURLValue = reportingURL {
+                            GleanMetrics.FxSuggest.reportingUrl.set(url: reportingURLValue)
+                        }
```

Comment:
nit, indentation
```suggestion
                    if let reportingURLValue = reportingURL {
                        GleanMetrics.FxSuggest.reportingUrl.set(url: reportingURLValue)
                    }
```

Reply:
You can also use the [`if let` shorthand](https://github.com/apple/swift-evolution/blob/7f69f7e8dd44060810e953ec4a5519231ee342b7/proposals/0345-if-let-shorthand.md) added in Swift 5.7!

```suggestion
                    if let reportingURL {
                        GleanMetrics.FxSuggest.reportingUrl.set(url: reportingURL)
                    }
```